interoperability-layer-openhim/swarm.sh

```diff
@@ -1,52 +1,67 @@
 #!/bin/bash
 
-# Constants
-readonly ACTION=$1
-readonly MODE=$2
-
-COMPOSE_FILE_PATH=$(
-  cd "$(dirname "${BASH_SOURCE[0]}")" || exit
-  pwd -P
-)
-readonly COMPOSE_FILE_PATH
-
-# Import libraries
-ROOT_PATH="${COMPOSE_FILE_PATH}/.."
-. "${ROOT_PATH}/utils/config-utils.sh"
-. "${ROOT_PATH}/utils/docker-utils.sh"
-. "${ROOT_PATH}/utils/log.sh"
-
-verify_core() {
-  local start_time
-  start_time=$(date +%s)
-  until [[ $(docker service ls -f name=instant_openhim-core --format "{{.Replicas}}") == *"${OPENHIM_CORE_INSTANCES}/${OPENHIM_CORE_INSTANCES}"* ]]; do
-    config::timeout_check "${start_time}" "openhim-core to start"
-    sleep 1
-  done
-
-  local await_helper_state
-  await_helper_state=$(docker service ps instant_await-helper --format "{{.CurrentState}}")
-  until [[ "${await_helper_state}" == *"Complete"* ]]; do
-    config::timeout_check "${start_time}" "openhim-core heartbeat check"
-    sleep 1
+declare ACTION=""
+declare MODE=""
+declare COMPOSE_FILE_PATH=""
+declare UTILS_PATH=""
+declare mongo_services=()
+declare service_names=()
+declare MONGO_CLUSTER_COMPOSE_FILENAME=""
+declare MONGO_DEV_COMPOSE_FILENAME=""
+declare OPENHIM_DEV_COMPOSE_FILENAME=""
+
+function init_vars() {
+  ACTION=$1
+  MODE=$2
+
+  COMPOSE_FILE_PATH=$(
+    cd "$(dirname "${BASH_SOURCE[0]}")" || exit
+    pwd -P
+  )
+
+  UTILS_PATH="${COMPOSE_FILE_PATH}/../utils"
+
+  mongo_services=(
+    "mongo-1"
+  )
+  if [[ "${NODE_MODE}" == "cluster" ]]; then
+    for i in {2..3}; do
+      mongo_services=(
+        "${mongo_services[@]}"
+        "mongo-$i"
+      )
+    done
+  fi
 
-    await_helper_state=$(docker service ps instant_await-helper --format "{{.CurrentState}}")
-    if [[ "${await_helper_state}" == *"Failed"* ]] || [[ "${await_helper_state}" == *"Rejected"* ]]; then
-      log error "Fatal: Received error when trying to verify state of openhim-core. Error:
-       $(docker service ps instant_await-helper --no-trunc --format '{{.Error}}')"
-      exit 1
-    fi
-  done
+  service_names=(
+    "${mongo_services[@]}"
+    "openhim-core"
+    "openhim-console"
+  )
+
+  readonly ACTION
+  readonly MODE
+  readonly COMPOSE_FILE_PATH
+  readonly UTILS_PATH
+  readonly mongo_services
+  readonly service_names
+}
 
-  try "docker service rm instant_await-helper" "Failed to remove await helper"
+# shellcheck disable=SC1091
+function import_sources() {
+  source "${UTILS_PATH}/docker-utils.sh"
+  source "${UTILS_PATH}/config-utils.sh"
+  source "${UTILS_PATH}/log.sh"
 }
 
-verify_mongos() {
-  log info 'Waiting to ensure all the mongo instances for the replica set are up and running'
+function verify_mongos() {
```

Comment:
Is verify_mongos not rendered redundant by docker::deploy_sanity?

Reply:
yes nice done :)